src/hooks/http.test.js

```diff
@@ -587,297 +469,20 @@ describe('http hook', () => {
-    const callback1 = jest.fn();
-
-    // Blacklisted.
-    const options1 = { host: edgeHost };
-    httpHook.httpRequestWrapper(originalRequestFn)(options1, callback1);
-    expect(originalRequestFn).toHaveBeenCalledWith(options1, callback1);
-
-    originalRequestFn.mockClear();
-
-    // Already traced.
-    const options3 = { host: 'bla.com' };
-    const callback3 = jest.fn();
-    callback3.__lumigoSentinel = true;
-    utils.getEdgeHost.mockReturnValueOnce(edgeHost);
-    httpHook.httpRequestWrapper(originalRequestFn)(options3, callback3);
-    expect(originalRequestFn).toHaveBeenCalledWith(options3, callback3);
-
-    originalRequestFn.mockClear();
-
-    // Error within Lumigo's code.
-    const options5 = { host: 'bla.com' };
-    const callback5 = jest.fn();
-    utils.getEdgeHost.mockReturnValueOnce(edgeHost);
-    originalRequestFn.mockImplementationOnce(() => {
-      throw new Error();
-    });
-
-    httpHook.httpRequestWrapper(originalRequestFn)(options5, callback5);
-    expect(originalRequestFn).toHaveBeenCalledTimes(2);
-    expect(originalRequestFn).toHaveBeenCalledWith(
-      options5,
-      expect.any(Function)
-    );
-    originalRequestFn.mockClear();
-
-    spies.randomBytes.mockReturnValueOnce(Buffer.from('aa'));
-    // Regular case
-    const options2 = {
-      host: 'baba.amazonaws.com',
-      port: 443,
-      protocol: 'https:',
-      path: '/api/where/is/satoshi',
-      method: 'POST',
-      headers: { X: 'Y' },
-    };
-
-    const callback2 = jest.fn();
-
-    const clientRequest = { a: 'b' };
-    originalRequestFn.mockReturnValueOnce(clientRequest);
-    utils.getEdgeHost.mockReturnValueOnce(edgeHost);
-
-    const expectedHeaders2 = {
-      X: 'Y',
-      ['X-Amzn-Trace-Id']:
-        'Root=1-00006161-64a1b06067c2100c52e51ef4;Parent=28effe37598bb622;Sampled=0',
-    };
-    const expectedOptions2 = Object.assign({}, options2, {
-      headers: expectedHeaders2,
-    });
-    expect(
-      httpHook.httpRequestWrapper(originalRequestFn)(options2, callback2)
-    ).toEqual(clientRequest);
-
-    expect(originalRequestFn).toHaveBeenCalledWith(
-      expectedOptions2,
-      expect.any(Function)
-    );
-
-    originalRequestFn.mockClear();
-    // No callback provided case
-    const options4 = {
-      host: 'asdf1.com',
-      port: 443,
-      protocol: 'https:',
-      path: '/api/where/is/satoshi',
-      method: 'POST',
-      headers: { X: 'Y' },
-    };
-
-    const clientRequest4 = { a: 'b' };
-    originalRequestFn.mockReturnValueOnce(clientRequest4);
-    utils.getEdgeHost.mockReturnValueOnce(edgeHost);
-    expect(httpHook.httpRequestWrapper(originalRequestFn)(options4)).toEqual(
-      clientRequest4
-    );
-
-    expect(originalRequestFn).toHaveBeenCalledWith(options4);
-
-    expect(spies.shimmer).toHaveBeenCalledWith(
-      clientRequest4,
-      'on',
-      expect.any(Function)
-    );
-
-    //Circular object
-    const a6 = {};
```

Comment:
you need to keep a circular test

Reply:
Yes, now its called `wrapHttp - circular object wrapper cutting object`